src/commands/generateEditorConfig.ts

```diff
@@ -0,0 +1,52 @@
+import {exists, writeFile} from 'fs';
+import {workspace, window} from 'vscode';
+import {Utils} from '../utils';
+
+/**
+ * Generate an .editorconfig file in the root of the workspace based on the current vscode settings.
+ */
+export function generateEditorConfig() {
+	if (!workspace.rootPath) {
+		window.showInformationMessage('Please open a folder before generating an .editorconfig file');
+		return;
+	}
+
+	const editorConfigurationNode = workspace.getConfiguration('editor');
+	const settings = Utils.toEditorConfig({
+		insertSpaces: editorConfigurationNode.get<string | boolean>('insertSpaces'),
+		tabSize: editorConfigurationNode.get<string | number>('tabSize')
+	});
+
+	let fileContents =
```

Comment:
Not sure about this one, it feels quite ugly. Think I rather prefer something like

```js
let fileContents = ['root = true', '', '*'];
```

And join them with `fileContents.join('\n');`